plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A class that highlights the user's
+ * viewport on the minimap.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+const blockEvents = new Set([
+  Blockly.Events.VIEWPORT_CHANGE,
+  Blockly.Events.BLOCK_CHANGE,
+  Blockly.Events.BLOCK_CREATE,
+  Blockly.Events.BLOCK_DELETE,
+  Blockly.Events.BLOCK_DRAG,
+  Blockly.Events.BLOCK_MOVE]);
+
+const borderRadius = 6;
+
+/**
+ * A class that highlights the user's viewport on the minimap.
+ */
+export class FocusMode {
```

Comment:
Optional: I don't love FocusMode as the name of this class - it feels weird to write "new FocusMode()", I think because FocusMode sounds like a state or boolean as opposed to a class responsible for drawing a highlighted region. Maybe something like FocusRegion or FocusRect?